### PR TITLE
Blog: Interacting with ESP32 Using Node-RED

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/11/esp32-with-node-red.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
+---
+title: Interacting with ESP32 Using Node-RED and MQTT
+subtitle: Building IoT Flows with ESP32 and FlowFuse
+description:  Learn how to connect your ESP32 with Node-RED using MQTT in this easy-to-follow guide. Build a user-friendly dashboard with FlowFuse Dashboard to control your IoT devices remotely. Ideal for beginners and IoT hobbyists!
+date: 2024-11-14
+authors: ["sumit-shinde"]
+image:
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/esp32-with-node-red.md:7]: Todo?

File: src/blog/2024/11/esp32-with-node-red.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+const char* mqtt_client_id = ""; // Replace with your MQTT client ID
+const char* mqtt_username = ""; // Replace with your MQTT username
+const char* mqtt_password = ""; // Replace with your MQTT password
+
+WiFiClient espClient;
+PubSubClient client(espClient);
+
+// Function to connect to WiFi
+void setup_wifi() {
+    delay(10);
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/esp32-with-node-red.md:157]: Curious as to why the delay?

File: src/blog/2024/11/esp32-with-node-red.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+The ESP32 is a powerful microcontroller with Wi-Fi and Bluetooth, commonly used for IoT applications. This tutorial demonstrates how to set up communication between the ESP32 and Node-RED using MQTT, along with an interactive dashboard via FlowFuse for a user-friendly interface.
+
+<!--more-->
+
+## What is ESP32?
+
+![ESP32](./images/esp32.webp){data-zoomable}
+_ESP32_
+
+The ESP32 is an affordable and powerful microchip that combines Wi-Fi and Bluetooth in one small package. It's commonly used in smart devices like home automation systems, wearables, and other IoT projects. Despite its low cost (around $6), it offers strong performance, and low power consumption, and is compatible with popular platforms like Arduino. Whether you're a hobbyist or a business, the ESP32 provides great value, making it easy to create wireless devices without a big investment.
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/esp32-with-node-red.md:23]: Let's combine this with the sentence above, and have the video under the `more` tag.

File: src/blog/2024/11/esp32-with-node-red.md (hunk 2 above)
Comment (sumitshinde-84) [reply src/blog/2024/11/esp32-with-node-red.md:150]: By adding a delay, we allow a brief moment for the serial communication to properly initialize before printing the first message. However, this delay is not strictly necessary we can remove it.

File: src/blog/2024/11/esp32-with-node-red.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    delay(10);
+    Serial.println();
+    Serial.print("Connecting to ");
+    Serial.print(ssid);
+    WiFi.begin(ssid, password);
+
+    while(WiFi.status() != WL_CONNECTED) {
+        delay(500);
+        Serial.print(".");
+ }
```
[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply src/blog/2024/11/esp32-with-node-red.md:159]: ?

File: src/blog/2024/11/esp32-with-node-red.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    Serial.print("Message arrived [");
+    Serial.print(topic);
+    Serial.print("]: ");
+
+ String msg;
+    for (int i = 0; i < length; i++) {
+ msg += (char)payload[i];
+ }
+    Serial.println(msg);
+
```
[the remark this answers is not in the document]
Comment (ZJvandeWeg) [reply src/blog/2024/11/esp32-with-node-red.md:177]: What error? it seems like an unneeded debug statement?

[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply src/blog/2024/11/esp32-with-node-red.md:177]: Do we want to remove the debug statement or this entire part of the code that you have added in the suggestion?

[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply src/blog/2024/11/esp32-with-node-red.md:177]: we are using the `msg` variable further in the code

[the remark this answers is not in the document]
Comment (ZJvandeWeg) [reply src/blog/2024/11/esp32-with-node-red.md:177]: I'd say remove the debug statement. Message colelction should remain indeed, my bad.

[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply src/blog/2024/11/esp32-with-node-red.md]: removed all unnecessary debug statements

**Pull request merged**